msal/src/Microsoft.Identity.Client/AuthenticationResult.cs

```diff
@@ -43,29 +43,62 @@ namespace Microsoft.Identity.Client
     public partial class AuthenticationResult
 #pragma warning restore CS1574 // XML comment has cref attribute that could not be resolved
     {
-        private const string Oauth2AuthorizationHeader = "Bearer ";
-        private readonly MsalAccessTokenCacheItem _msalAccessTokenCacheItem;
-        private readonly MsalIdTokenCacheItem _msalIdTokenCacheItem;
-
+        private const string Oauth2AuthorizationHeader = "Bearer ";
+
+        /// <summary>
+        /// Constructor meant for test purposes, to allow mocking of authentication flows. Developers should 
+        /// never new up <see cref="AuthenticationResult"/> in product code.
+        /// </summary>
+        public AuthenticationResult(

```

Comment:
I know it's a pretty horrible constructor, but I'd rather keep the object immutable than adding setters

Reply:
For MSAL (since we _are_ still in preview) can we just change this to an interface and break it instead of living with this in GA back-compat?